src/test/streaming_test.cpp

```diff
@@ -618,33 +688,109 @@ node {
-// [x] Send 1 request, receive Y responses (sync client side)
+TEST_F(StreamingTest, ManualTimestampNotInRange) {
+    const std::string pbTxt{R"(
+input_stream: "in"
+output_stream: "out"
+node {
+  calculator: "AddOneSingleStreamTestCalculator"
+  input_stream: "in"
+  output_stream: "out"
+}
+    )"};
+    ::mediapipe::CalculatorGraphConfig config;
+    ASSERT_TRUE(::google::protobuf::TextFormat::ParseFromString(pbTxt, &config));
 
-// [x] Automatic timestamping
-// [x] Manual timestamping
+    MediapipeGraphExecutor executor{
+        this->name, this->version, config,
+        {{"in", mediapipe_packet_type_enum::OVTENSOR}},
+        {{"out", mediapipe_packet_type_enum::OVTENSOR}},
+        {"in"}, {"out"}, {}};
 
-// Negative:
-// [:] Calculators throwing - MP does not allow catching
-// [x] Error during graph initialization (bad pbtxt)
-// [x] Error installing observer (wrong outputName)
-// [x] Timestamp not an int64 ?
-// [:] Error during serialization - cannot write test, wrong serialization type=death instead of exception
-// [:] Error during startrun - cannot write test, startrun cannot fail
-// [x] Error during graph execution - Process() returning non Ok?
-// [x] Error during first deserialization
-// [x] Error during subsequent deserializations
-// [:] Error when closing all packet sources - cannot, this function never fails
-// [x] Error waiting until done (this will return any an error during execution - has list of errors)
-// [x] Error when writing to disconnected client
-// [x] Wrong timestamping (non monotonous) on client side
+    // Timestamps not allowed in stream
+    // Expect continuity of operation and response with error message
+    for (auto timestamp : std::vector<int64_t>{
+        ::mediapipe::kint64min,     // ::mediapipe::Timestamp::Unset()
+        ::mediapipe::kint64min + 1, // ::mediapipe::Timestamp::Unstarted()
+        ::mediapipe::kint64min + 2, // ::mediapipe::Timestamp::PreStream()
+        ::mediapipe::kint64max - 2, // ::mediapipe::Timestamp::PostStream()
+        ::mediapipe::kint64max - 1, // ::mediapipe::Timestamp::OneOverPostStream()
+        ::mediapipe::kint64max,     // ::mediapipe::Timestamp::Done()
+    }) {
+        std::mutex mtx;
+        prepareRequest(this->firstRequest, {{"in", 3.5f}}, timestamp);
+        EXPECT_CALL(this->stream, Read(_))
+            .WillOnce(DisconnectWhenNotified(mtx));
+        EXPECT_CALL(this->stream, Write(_, _))
+            .WillOnce(SendErrorAndNotifyEnd(
+                Status(StatusCode::MEDIAPIPE_INVALID_TIMESTAMP).string() 
+                    + std::string{" - "} 
+                    + ::mediapipe::Timestamp::CreateNoErrorChecking(timestamp).DebugString() 
+                    + std::string{"; partial deserialization of first request"},
+                mtx));
+        ASSERT_EQ(executor.inferStream(this->firstRequest, this->stream), StatusCode::OK);
+    }
+
+    // Allowed in stream
```

Comment:
Test name is  ManualTimestampNotInRange.
I wonder if having both positive and negative cases in a single test is a good idea.

Reply:
It is split now